crates/tabby-common/src/api/event.rs

```diff
@@ -92,8 +93,9 @@ pub struct Segments {
     pub clipboard: Option<String>,
 }
 
+#[async_trait]
 pub trait EventLogger: Send + Sync {
-    fn log(&self, e: Event);
+    async fn log(&self, e: Event);
```

Comment:
keep this as non-async function.

Example: https://github.com/TabbyML/tabby/blob/7ba5e72e6a5bf92dd17050f5eaecc5eedd1eecba/ee/tabby-webserver/src/hub/api.rs#L78